File: 3_laboratorijska_vjezba/1_zadatak/parrot.c
#include<stdio.h>
#include<stdlib.h>

typedef const char* (*PTRFUN)();

struct Parrot
{
    PTRFUN* vtable;
    const char* ime;
};

const char* name(void* animal) {
    return ((struct Parrot*)animal)->ime;
}

const char* greet() {
    return "gaga";
}

const char* menu() {
    return "tigrove";
}

PTRFUN parrot_table[3] = {*name, *greet, *menu};

void* create(const char* ime) {
    struct Parrot* parrot = (struct Parrot*)malloc(sizeof(struct Parrot));

    parrot->ime = ime;
    parrot->vtable = parrot_table;

    return parrot;
}

File: 3_laboratorijska_vjezba/1_1_zadatak/Zadaca/src/main/java/ooup/plugins/Parrot.java
package ooup.plugins;

import ooup.Animal;

public class Parrot extends Animal {
    private String name;

    public Parrot(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "dagadadagada";
    }

    @Override
    public String menu() {
        return "meat";
    }
}
